### ISSUE TITLE
Cannot use non-standard URL

### ISSUE THREAD

**Issue by jalanb** [coddingtonbear/jirafs#29]
jirafs expects a URL like `http://my.jira.server/browse/...`. However our JIRA server is set up to use `http://my.jira.server/jira/browse/...` and jirfs barfs on that (below)

I put in a quick fix, changing [line 213 of ticketfolder.py](https://github.com/coddingtonbear/jirafs/blob/development/jirafs/ticketfolder.py#L213) to 
```python
        match = re.match('\/.*\/browse\/(\w+-\d+)\/?.*', parts.path)
```
However this was insufficient as later on the code does not know to use the `/jira` part in constructing URLs

Traceback:

```shell
$ jirafs clone https://server.com/jira/browse/STUFF-999
Traceback (most recent call last):
  File "/home/jalanb/bin/jirafs", line 11, in <module>
    sys.exit(main())
  File "/home/jalanb/lib/python2.7/site-packages/jirafs/cmdline.py", line 86, in main
    extra, jira=jira, path=os.getcwd(), command_name=command_name
  File "/home/jalanb/lib/python2.7/site-packages/jirafs/plugin.py", line 172, in execute_command
    result = cmd.handle(**kwargs)
  File "/home/jalanb/lib/python2.7/site-packages/jirafs/commands/clone.py", line 37, in handle
    return self.clone(path, ticket_url, jira)
  File "/home/jalanb/lib/python2.7/site-packages/jirafs/commands/clone.py", line 175, in clone
    jira,
  File "/home/jalanb/lib/python2.7/site-packages/jirafs/commands/clone.py", line 44, in clone_from_issue
    folder = TicketFolder.initialize_ticket_folder(ticket_url, path, jira)
  File "/home/jalanb/lib/python2.7/site-packages/jirafs/ticketfolder.py", line 443, in initialize_ticket_folder
    (instance.ticket_number, instance.path, )
  File "/home/jalanb/lib/python2.7/site-packages/jirafs/ticketfolder.py", line 216, in ticket_number
    "Could not infer ticket number from URL %s" % self.issue_url
ValueError: Could not infer ticket number from URL https://server.com/jira/browse/STUFF-999
```

**Comment by coddingtonbear** [comment]
Oh, neat -- non-standard URLs never occurred to me.  I have some ideas and will try to have a look at this tonight.

**Comment by coddingtonbear** [comment]
Hey there, @jalanb -- I'm not sure how to verify when I have this working properly with any sort of confidence aside from trying to set up my own instance using a non-standard URL, and I wonder if you could help me out.  It's kind of a long-shot, but is the server you're cloning from publicly-readable? If not, I think I might still be able to get a little further if I knew what the next exception you encountered was after changing the regex above.

The reason that this is a little tricky is that we're relying upon [the python-jira library](http://pythonhosted.org/jira/) for interacting with JIRA, and -- although I only really had a couple of minutes to look into it -- I'm suspicious that the problem might be there rather than in Jirafs.

**Comment by jalanb** [comment]
Unfortunately our server is private

Not at work ATM, so more detail tomorrow, but the next error after changing the regex was when jirafs tried to GET json for the issue I had given on command line, and got an HTML page instead. Latter was probably a verbose "Not found" message. Shall re-try tomorrow

**Comment by jalanb** [comment]
After tweaking the regexp I ran the jirafs command to clone an issue. This gave back html wrapped in an error message. The html looks like our (wiki) landing page, so I've removed the majority of it below
```shell
$ jirafs clone https://server.com/jira/browse/STUFF-999
ERROR:root:No JSON object could be decoded
<!DOCTYPE html>
<html>
<head>
    <title>Welcome to the ZZZ Wiki - ZZZ - ZZZ Wiki</title>
</head>
<body             onload="placeFocus()"
     id="com-atlassian-confluence" class="theme-documentation  aui-theme-default aui-layout">
    <div id="full-height-container">
    ...
                    </div>
<header id="header" role="banner">
    ...
</header>
...
</body>
</html>
Traceback (most recent call last):
  File "/home/jalanb/bin/jirafs", line 11, in <module>
    sys.exit(main())
  File "/home/jalanb/lib/python2.7/site-packages/jirafs/cmdline.py", line 86, in main
    extra, jira=jira, path=os.getcwd(), command_name=command_name
  File "/home/jalanb/lib/python2.7/site-packages/jirafs/plugin.py", line 172, in execute_command
    result = cmd.handle(**kwargs)
  File "/home/jalanb/lib/python2.7/site-packages/jirafs/commands/clone.py", line 37, in handle
    return self.clone(path, ticket_url, jira)
  File "/home/jalanb/lib/python2.7/site-packages/jirafs/commands/clone.py", line 175, in clone
    jira,
  File "/home/jalanb/lib/python2.7/site-packages/jirafs/commands/clone.py", line 44, in clone_from_issue
    folder = TicketFolder.initialize_ticket_folder(ticket_url, path, jira)
  File "/home/jalanb/lib/python2.7/site-packages/jirafs/ticketfolder.py", line 443, in initialize_ticket_folder
    (instance.ticket_number, instance.path, )
  File "/home/jalanb/lib/python2.7/site-packages/jirafs/ticketfolder.py", line 923, in log
    self.issue,
  File "/home/jalanb/lib/python2.7/site-packages/jirafs/ticketfolder.py", line 232, in issue
    self._issue = self.jira.issue(self.ticket_number)
  File "/home/jalanb/lib/python2.7/site-packages/jirafs/ticketfolder.py", line 225, in jira
    config=self.get_config()
  File "/home/jalanb/lib/python2.7/site-packages/jirafs/utils.py", line 253, in <lambda>
    return lambda domain, config=None: get_jira(domain, config)
  File "/home/jalanb/lib/python2.7/site-packages/jirafs/utils.py", line 233, in get_jira
    jira = JIRA(login_data, basic_auth=basic_auth)
  File "/home/jalanb/lib/python2.7/site-packages/jira/client.py", line 264, in __init__
    si = self.server_info()
  File "/home/jalanb/lib/python2.7/site-packages/jira/client.py", line 1620, in server_info
    return self._get_json('serverInfo')
  File "/home/jalanb/lib/python2.7/site-packages/jira/client.py", line 2041, in _get_json
    raise e
ValueError: No JSON object could be decoded
```

**Comment by coddingtonbear** [comment]
Fixed in 8cd56b47546610dfca15d5f325431227db62ec4d!  Let me know if you're still having trouble with this (assuming you're still interested in giving it a shot, of course, @jalanb).

Cheers!

**Comment by jalanb** [comment]
I've moved on, and no longer have access to Jira

However, @jimlawton - you may wish to give jirafs another try now
